.github/workflows/ci.yml

```diff
@@ -43,10 +43,39 @@ jobs:
           ruby-version: "3.0"
           bundler-cache: true # runs 'bundle install' and caches installed gems automatically
 
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+        include:
+          - platform: aarch64-linux
+        with:
+          install: true
+
+      - name: Enable Multiarch
+        run: |
+          docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+        include:
+          - platform: aarch64-linux
+
+      - name: Create builder
+        run: |
+          docker buildx create --name mbuilder
+          docker buildx use mbuilder
+          docker buildx inspect
+          docker buildx ls
+        include:
+          - platform: aarch64-linux
+
+      - name: Export Docker Buildx arguments
+        run: |
+          echo "DOCKER_BUILDX_ARGS=--platform=linux/arm64" >> $GITHUB_ENV
+        include:
+          - platform: aarch64-linux
+
       - name: Build docker image
         run: |
           docker buildx create --driver docker-container --use
-          bundle exec rake build:${PLATFORM} RCD_DOCKER_BUILD="docker buildx build --cache-from=type=local,src=tmp/build-cache --cache-to=type=local,dest=tmp/build-cache-new --load"
+          bundle exec rake build:${PLATFORM} RCD_DOCKER_BUILD="docker buildx build ${DOCKER_BUILDX_ARGS} --cache-from=type=local,src=tmp/build-cache --cache-to=type=local,dest=tmp/build-cache-new --load"
```

Comment:
I guess this doesn't help local developers trying to run `bundle exec rake build`. Maybe we should push this change down into the `Rakefile`?

Reply:
Not sure why I'm getting this locally: 

```
$ bundle exec rake build:aarch64-linux  RCD_DOCKER_BUILD="docker buildx build --platform=linux/arm64 --cache-from=type=local,src=tmp/build-cache --cache-to=type=local,dest=tmp/build-cache-new --load"
docker buildx build --platform=linux/arm64 --cache-from=type=local,src=tmp/build-cache --cache-to=type=local,dest=tmp/build-cache-new --load -f tmp/docker/common-Dockerfile.mri.aarch64-linux .
[+] Building 3.1s (11/38)
 => [internal] load build definition from common-Dockerfile.mri.aarch64-linux                                                                                                                                                                0.0s
 => => transferring dockerfile: 6.85kB                                                                                                                                                                                                       0.0s
 => [internal] load .dockerignore                                                                                                                                                                                                            0.0s
 => => transferring context: 2B                                                                                                                                                                                                              0.0s
 => [internal] load metadata for quay.io/pypa/manylinux2014_aarch64:latest                                                                                                                                                                   0.1s
 => [internal] load build context                                                                                                                                                                                                            0.0s
 => => transferring context: 663B                                                                                                                                                                                                            0.0s
 => [ 1/34] FROM quay.io/pypa/manylinux2014_aarch64@sha256:8fcd071d89dab8043aaf97360662d84833c2af54cb65bffcf05cd0ed2fda4b06                                                                                                                  0.0s
 => => resolve quay.io/pypa/manylinux2014_aarch64@sha256:8fcd071d89dab8043aaf97360662d84833c2af54cb65bffcf05cd0ed2fda4b06                                                                                                                    0.0s
 => CACHED [ 2/34] RUN yum install -y autoconf gcc-c++ libtool readline-devel sqlite-devel ruby openssl-devel xz cmake sudo less libffi-devel git wget                                                                                       0.0s
 => CACHED [ 3/34] RUN rm -f /usr/local/bin/sudo &&     groupadd -r sudo &&     echo "%sudo  ALL=(ALL)       ALL" >> /etc/sudoers                                                                                                            0.0s
 => CACHED [ 4/34] RUN groupadd -r rvm && useradd -r -g rvm -G sudo -p "" --create-home rvm                                                                                                                                                  0.0s
 => CACHED [ 5/34] RUN echo "source /etc/profile.d/rvm.sh" >> /etc/rubybashrc &&     echo "source /etc/rubybashrc" >> /etc/bashrc &&     echo "source /etc/rubybashrc" >> /etc/bash.bashrc                                                   0.0s
 => CACHED [ 6/34] RUN mkdir ~/.gnupg &&     chmod 700 ~/.gnupg &&     echo "disable-ipv6" >> ~/.gnupg/dirmngr.conf                                                                                                                          0.0s
 => ERROR [ 7/34] RUN gpg --keyserver hkp://keyserver.ubuntu.com --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB &&     (curl -L http://get.rvm.io | sudo bash) &&     bash -c "         sour  2.6s
------
 > [ 7/34] RUN gpg --keyserver hkp://keyserver.ubuntu.com --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB &&     (curl -L http://get.rvm.io | sudo bash) &&     bash -c "         source /etc/rubybashrc &&         rvm autolibs disable &&         rvmsudo rvm cleanup all ":
#0 0.225 gpg: keyring `/home/rvm/.gnupg/secring.gpg' created
#0 0.229 gpg: keyring `/home/rvm/.gnupg/pubring.gpg' created
#0 0.237 gpg: requesting key D39DC0E3 from hkp server keyserver.ubuntu.com
#0 0.237 gpg: requesting key 39499BDB from hkp server keyserver.ubuntu.com
#0 1.260 gpg: /home/rvm/.gnupg/trustdb.gpg: trustdb created
#0 1.274 gpg: key D39DC0E3: public key "Michal Papis (RVM signing) <mpapis@gmail.com>" imported
#0 1.288 gpg: key 39499BDB: public key "Piotr Kuczynski <piotr.kuczynski@gmail.com>" imported
#0 1.303 gpg: no ultimately trusted keys found
#0 1.304 gpg: Total number processed: 2
#0 1.304 gpg:               imported: 2  (RSA: 2)
#0 1.407 /etc/rubybashrc: line 1: /etc/profile.d/rvm.sh: No such file or directory
#0 1.417   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
#0 1.423                                  Dload  Upload   Total   Spent    Left  Speed
100   194  100   194    0     0    607      0 --:--:-- --:--:-- --:--:--   664
 16 24535   16  4113    0     0   3621      0  0:00:06  0:00:01  0:00:05 27604
#0 2.529 curl: (23) Failed writing body (1354 != 1371)
------
WARNING: local cache import at tmp/build-cache not found due to err: could not read tmp/build-cache/index.json: open tmp/build-cache/index.json: no such file or directory
common-Dockerfile.mri.aarch64-linux:26
--------------------
  25 |     # install rvm, RVM 1.26.0+ has signed releases, source rvm for usage outside of package scripts
  26 | >>> RUN gpg --keyserver hkp://keyserver.ubuntu.com --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB && \
  27 | >>>     (curl -L http://get.rvm.io | sudo bash) && \
  28 | >>>     bash -c " \
  29 | >>>         source /etc/rubybashrc && \
  30 | >>>         rvm autolibs disable && \
  31 | >>>         rvmsudo rvm cleanup all "
  32 |
--------------------
error: failed to solve: process "/bin/sh -c gpg --keyserver hkp://keyserver.ubuntu.com --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB &&     (curl -L http://get.rvm.io | sudo bash) &&     bash -c \"         source /etc/rubybashrc &&         rvm autolibs disable &&         rvmsudo rvm cleanup all \"" did not complete successfully: exit code: 1
rake aborted!
```

Reply:
It works if I save the `curl` output to a file:

```shell
curl -L -o /tmp/rvm.sh
bash /tmp/rvm.sh
```

Reply:
Oddly `sudo` is not working on arm64:

```
$ docker run --platform linux/arm64 -it quay.io/pypa/manylinux2014_aarch64
<snip>
[root@6462ad36c7cf /]#  rm -f /usr/local/bin/sudo && \
>     groupadd -r sudo && \
>     echo "%sudo  ALL=(ALL)       ALL" >> /etc/sudoers
[root@6462ad36c7cf /]#  groupadd -r rvm && useradd -r -g rvm -G sudo -p "" --create-home rvm
[root@6462ad36c7cf /]# su rvm
sud[rvm@6462ad36c7cf /]$ sudo
Error while loading /usr/bin/sudo: Permission denied
[rvm@6462ad36c7cf /]$ ls -al /usr/bin/sudo
---s--x--x 1 root root 204648 Oct 14 12:43 /usr/bin/sudo
```

Works fine on x86:

```
$ docker exec run -it quay.io/pypa/manylinux2014_x86_64 bash
<snip>
[rvm@91b718d778e2 /]$ which sudo
/opt/rh/devtoolset-10/root/usr/bin/sudo
[rvm@91b718d778e2 /]$ ls -al /usr/bin/sudo
---s--x--x 1 root root 151424 Oct 14 12:28 /usr/bin/sudo
[rvm@91b718d778e2 /]$ sudo echo "hi"
hi
```

Reply:
On my Apple M1, the same `sudo` command works fine with the `quay.io/pypa/manylinux2014_aarch64` image. This makes me wonder if this is just a QEMU issue.

Reply:
https://bbs.archlinux.org/viewtopic.php?pid=1822527#p1822527 and https://unix.stackexchange.com/a/640038 might be an explanation why `sudo` doesn't work under QEMU.

Reply:
@flavorjones Do we really need to run these builds as the `rvm` user? Can we get rid of `sudo`?

Reply:
9537a81 should enable the `binfmt_misc` OC flags to make `sudo` work under `aarch64-linux`.

Reply:
https://github.com/rake-compiler/rake-compiler-dock/runs/5985250394?check_suite_focus=true is running, but it's pretty slow. Feels like we should be using a native arm64 machine.

Reply:
@stanhu It's a good question (about sudo). I haven't been present for the whole lifetime of RCD, but the use of an `rvm` user dates back to 66f33ef. That commit also introduced most of the usage of `sudo` in the Dockerfile. I'm not sure why it's not being run by the `root` account, maybe @larskanis remembers?